src/BenchmarksApps/TodosApi/Program.cs

```diff
@@ -0,0 +1,76 @@
+using Microsoft.Extensions.Logging.Configuration;
+using Npgsql;
+using TodosApi;
+
+var builder = WebApplication.CreateSlimBuilder(args);
+
+// Load custom configuration
+var settingsFiles = new[] { "appsettings.json", $"appsettings.{builder.Environment.EnvironmentName}.json" };
+foreach (var settingsFile in settingsFiles)
+{
+    builder.Configuration.AddJsonFile(builder.Environment.ContentRootFileProvider, settingsFile, optional: true, reloadOnChange: true);
+}
+
+#if DEBUG || DEBUG_DATABASE
+builder.Configuration.AddUserSecrets<Program>();
+#endif
+
+#if ENABLE_LOGGING
+// Configure logging
+builder.Logging
+    .AddConfiguration(builder.Configuration.GetSection("Logging"))
+    .AddSimpleConsole();
+#endif
+
+// Configure authentication & authorization
+builder.Services.AddAuthentication()
+    .AddJwtBearer(JwtConfiguration.ConfigureJwtBearer(builder));
+
+builder.Services.AddAuthorization();
+
+// Configure data access
+var connectionString = builder.Configuration.GetConnectionString("TodoDb")
+    ?? builder.Configuration["CONNECTION_STRING"]
+    ?? throw new InvalidOperationException("""
+        Connection string not found.
+        If running locally, set the connection string in user secrets for key 'ConnectionStrings:TodoDb'.
+        If running after deployment, set the connection string via the environment variable 'CONNECTIONSTRINGS__TODODB'.
+        """);
+builder.Services.AddSingleton(_ => new NpgsqlSlimDataSourceBuilder(connectionString).Build());
+
+// Configure JSON serialization
+builder.Services.ConfigureHttpJsonOptions(options =>
+{
+    options.SerializerOptions.TypeInfoResolverChain.Insert(0, TodoApiJsonSerializerContext.Default);
+});
+
+// Configure health checks
+builder.Services.AddHealthChecks()
+    .AddCheck<DatabaseHealthCheck>("Database")
+    .AddCheck<JwtHealthCheck>("JwtAuthentication");
+
+// Problem details
+builder.Services.AddProblemDetails();
+
+var app = builder.Build();
+
+await Database.Initialize(app.Services, app.Logger);
+
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler();
+}
+
+app.MapHealthChecks("/health");
+
+app.MapTodoApi();
+
+// Enables testing request exception handling behavior
+app.MapGet("/throw", (HttpContext httpContext) => throw new InvalidOperationException("You hit the throw endpoint"));
```

Comment:
```suggestion
app.MapGet("/throw", () => throw new InvalidOperationException("You hit the throw endpoint"));
```

is that needed?

Reply:
I was getting a compilation error without it, which might be a RDG issue?

Reply:
https://github.com/dotnet/aspnetcore/issues/47275